c2rust-transpile/src/build_files/mod.rs

```diff
@@ -169,10 +169,8 @@ fn convert_module_list(
             // Don't add binary modules to lib.rs, these are emitted to
             // standalone, separate binary modules.
             false
-        } else if !is_binary && module_subset == ModuleSubset::Binaries {
-            false
         } else {
-            true
+            is_binary || !(module_subset == ModuleSubset::Binaries)
         }
```

Comment:
Personally I think this is much more readable as a match:
```rust
	match (is_binary, module_subset) {
		// Don't add binary modules to lib.rs, these are emitted to
		// standalone, separate binary modules.
		(true, ModuleSubset::Libraries) => false,
		(false, ModuleSubset::Binaries) => false,
		_ => true,
	}
```

Reply:
I think I found a better solution: `is_binary == is_mod_binary`.

Reply:
I like this, with the caveat that `is_mod_binary` is a misleading name--that name sounds like it indicates whether a module is binary, but the variable really indicates whether the module subset selected is the binary subset, so I would name it something like `is_binary_subset` or `want_binary`.

Reply:
Renamed in https://github.com/immunant/c2rust/pull/462/commits/c0610f2e65c5aae246529b06cc9b0b3375df22db.